### PR TITLE
Feature/get debug view value processing

### PR DESCRIPTION
**Pull request by oskrabanek**
Implementation of API proposal #60065 based on the changes proposed on the review
@safern

### CODE REVIEW

**Comment by safern** [comment]
Could you also update: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.Abstractions/ref/Microsoft.Extensions.Configuration.Abstractions.cs to add the new APIs? 

Here are instructions on how to do that: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/updating-ref-source.md

**Review by safern — COMMENTED** [review]
@Andree643 looks good overall! Could you also please add some tests for the new API?

**Comment by eerhardt** [comment]
Hi @Andree643 - do you have an idea when the tests for the new API will be added? Let us know if you are interested in pushing this PR forward.

**Comment by Andree643** [comment]
> Hi @Andree643 - do you have an idea when the tests for the new API will be added? Let us know if you are interested in pushing this PR forward.
Hi @eerhardt. I'll do my best to make it next week. I have some issues with building the solution and trying to resolve it.

File: src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationRootExtensions.cs [hunk 1]
```diff
@@ -17,6 +18,22 @@ public static class ConfigurationRootExtensions
         /// </summary>
         /// <returns> The debug view. </returns>
         public static string GetDebugView(this IConfigurationRoot root)
+        {
+            return GetDebugView(root, null);
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationRootExtensions.cs:22]:
```suggestion
            return GetDebugView(root, processValue: null);
```

File: src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationDebugViewContext.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        /// </summary>
+        public string Key { get; }
+
+        /// <summary>
+        /// Value of the current item
+        /// </summary>
+        public string? Value { get; }
+
+        /// <summary>
+        /// Provider used to get the value of the current item
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationDebugViewContext.cs:35]:
```suggestion
        /// Gets the <see cref="IConfigurationProvider" /> that was used to get the value of the current item.
```

**Review by eerhardt — COMMENTED** [review]
Looks pretty good. I just had some nits on some doc comments. Once those are addressed, I believe this can be merged.

**Pull request merged**